src/hooks/useOnceAtATime.ts

```diff
@@ -0,0 +1,42 @@
+import { useRef, useCallback } from 'react';
+
+/**
+ * useOnceAtATime
+ *
+ * A React hook that ensures an async function is only executed once at a time.
+ * While the function is in-flight, all subsequent calls return the same Promise.
+ * Once the function resolves or rejects, it can be called again.
+ *
+ * Useful for deduplicating concurrent requests (e.g. token validation, lazy loading).
+ *
+ * @template TArgs - Argument types of the async function
+ * @template TResult - Return type of the async function
+ *
+ * @param fn - The async function to guard
+ * @returns An object with:
+ *   - run: the deduplicated function
+ *   - isRunning: boolean indicating whether the function is currently executing
+ */
```

Comment:
I had to use this with `verifyToken` to solve a race condition I was facing where a second invocation thought I wasn't logged in even though I was.